src/builtins/BuiltinTemporal.cpp

```diff
@@ -206,6 +390,85 @@ static Value builtinTemporalCalendarPrototypeDay(ExecutionState& state, Value th
+    ASSERT(std::isfinite(epochDays.asNumber()));
+
+    return Value(TemporalCalendar::dayOfYear(state, epochDays));
+}
+
+static Value builtinTemporalCalendarPrototypeWeekOfYear(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalCalendar(state, thisValue, argc);
+
+    TemporalPlainDate* temporalDate = TemporalPlainDate::toTemporalDate(state, argv[0]).asObject()->asTemporalPlainDateObject();
+    return Value(TemporalCalendar::toISOWeekOfYear(state, temporalDate->year(), temporalDate->month(), temporalDate->day()));
+}
+
+static Value builtinTemporalCalendarPrototypeDaysInWeek(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalCalendar(state, thisValue, argc);
+
+    TemporalPlainDate::toTemporalDate(state, argv[0]).asObject()->asTemporalPlainDateObject();
+    return Value(7);
+}
+
+static Value builtinTemporalCalendarPrototypeDaysInMonth(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalCalendar(state, thisValue, argc);
+
+    TemporalPlainDate* temporalDate;
+
+    if (!argv[0].isObject() || !(argv[0].asObject()->isTemporalPlainDateObject() || argv[0].asObject()->isTemporalPlainDateTimeObject() || argv[0].asObject()->isTemporalYearMonthObject())) {
+        temporalDate = TemporalPlainDate::toTemporalDate(state, argv[0]).asObject()->asTemporalPlainDateObject();
+    } else {
+        temporalDate = argv[0].asObject()->asTemporalPlainDateObject();
+    }
+
+    return TemporalCalendar::ISODaysInMonth(state, temporalDate->year(), temporalDate->month());
+}
+
+static Value builtinTemporalCalendarPrototypeDaysInYear(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalCalendar(state, thisValue, argc);
+
+    TemporalPlainDate* temporalDate;
+
+    if (!argv[0].isObject() || !(argv[0].asObject()->isTemporalPlainDateObject() || argv[0].asObject()->isTemporalPlainDateTimeObject() || argv[0].asObject()->isTemporalYearMonthObject())) {
+        temporalDate = TemporalPlainDate::toTemporalDate(state, argv[0]).asObject()->asTemporalPlainDateObject();
+    } else {
+        temporalDate = argv[0].asObject()->asTemporalPlainDateObject();
+    }
+
+    return Value(DateObject::daysInYear(temporalDate->year()));
+}
+
+static Value builtinTemporalCalendarPrototypeMonthsInYear(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalCalendar(state, thisValue, argc);
+
+    if (!argv[0].isObject() || !(argv[0].asObject()->isTemporalPlainDateObject() || argv[0].asObject()->isTemporalPlainDateTimeObject() || argv[0].asObject()->isTemporalYearMonthObject())) {
+        TemporalPlainDate::toTemporalDate(state, argv[0]).asObject()->asTemporalPlainDateObject();
```

Comment:
There are many similar conversions and it is unclear for me whether they are doing anything. If not, these expressions can be simplified.

Reply:
[Here](https://tc39.es/proposal-temporal/#sec-temporal.calendar.prototype.monthsinyear) it says it's needed to be called. I think it's needed to be called because it will throw error if the first argument is bad.

Reply:
Do you need the `->asTemporalPlainDateObject();` part as well?

Reply:
No will delete it.